### PR TITLE
Feature/add sync destination username filter

### PR DESCRIPTION
**Pull request by peterbriggs42**
### Description of Change ###

Add a filter class to allow users to filter file event query results by sync destination username. 

### Issues Resolved ###
- closes #INTEG-1703

### Testing Procedure ###
```
>>> from py42.sdk.queries.fileevents.file_event_query import FileEventQuery
>>> from py42.sdk.queries.fileevents.filters.exposure_filter import SyncDestinationUsername
>>> filter = SyncDestinationUsername.is_in(['*@gmail.com'])
>>> results = sdk.securitydata.search_file_events(FileEventQuery.all(filter))
>>> [e['syncDestinationUsername'][0] for e in results.data['fileEvents']]
```

### PR Checklist ###
Did you remember to do the below?

- [x] Add unit tests to verify this change
- [x] Add an entry to CHANGELOG.md describing this change
- [x] Add docstrings for any new public parameters / methods / classes

### CODE REVIEW

**Comment by unparalleled-js** [comment]
Could you add a CHANGELOG entry for the new filter class?

File: tests/sdk/queries/fileevents/filters/test_exposure_filter.py [hunk 1]
```diff
@@ -411,6 +412,44 @@ def test_sync_destination_name_not_in_str_gives_correct_json_representation():
...
+
+
+def test_sync_destination_username_not_exists_str_gives_correct_json_representation():
+    _filter = SyncDestinationUsername.not_exists()
+    expected = NOT_EXISTS.format("syncDestinationUsername")
+    assert str(_filter) == expected
+
+
+def test_sync_destination_username_eq_str_gives_correct_json_representation():
+    _filter = SyncDestinationUsername.eq("test_user@gmail.com")
```
Comment (unparalleled-js) [inline tests/sdk/queries/fileevents/filters/test_exposure_filter.py:428]: It makes me nervous having real domains (in this case, for email addresses) in tests.  Can we just always use `@example.com`. I know nothing can really go wrong here.

`test_user@gmail.com` is definitely an account out in the wild somewhere

File: src/py42/sdk/queries/fileevents/filters/exposure_filter.py [hunk 2]
```diff
@@ -124,6 +124,14 @@ def choices():
         return get_attribute_keys_from_class(SyncDestination)
 
 
+class SyncDestinationUsername(FileEventFilterStringField):
+    """Class that filters events based on the name of the username associated with the cloud service
+    the file is synced with (applies to ``synced to cloud service`` events only).
```
Comment (unparalleled-js) [inline src/py42/sdk/queries/fileevents/filters/exposure_filter.py:129]: I don't see this listed in the UI under `EventType`. Is this one or what is this? Re: `synced to cloud service`

File: tests/sdk/queries/fileevents/filters/test_exposure_filter.py (hunk 1 above)
Comment (peterbriggs42) [reply tests/sdk/queries/fileevents/filters/test_exposure_filter.py:428]: Ahh that's a good point. Changed all real domain names to `example.com`

File: src/py42/sdk/queries/fileevents/filters/exposure_filter.py (hunk 2 above)
Comment (peterbriggs42) [reply src/py42/sdk/queries/fileevents/filters/exposure_filter.py:129]: I believe this refers to the value of `Exposure Type` in the UI. There's a value in the UI for "Synced to cloud service"

File: src/py42/sdk/queries/fileevents/filters/exposure_filter.py [hunk 3]
```diff
@@ -124,6 +124,14 @@ def choices():
         return get_attribute_keys_from_class(SyncDestination)
 
 
+class SyncDestinationUsername(FileEventFilterStringField):
+    """Class that filters events based on the name of the username associated with the cloud service
```
Comment (timabrmsn) [inline src/py42/sdk/queries/fileevents/filters/exposure_filter.py:128]: probably don't need to say `name of the username`, and could just say `username`

**Review by timabrmsn — APPROVED** [review]

**Pull request merged**